File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/shop/order/clientUpdateOrderRequest.ts
import { DeliveryMethod, Order, PaymentMethod } from "../../../..";

export interface ClientUpdateOrderRequest {
    id: number,
    contactClientName: string,
    contactPhone: string,
    deliveryAddress: string,
    deliveryTime: Date,
    paymentMethod: PaymentMethod,
    deliveryMethod: DeliveryMethod,
}

export function mapOrderToClientUpdateOrderRequest(order: Order): ClientUpdateOrderRequest {
    return {
        id: order.id,
        contactClientName: order.contactClientName,
        contactPhone: order.contactPhone,
        deliveryAddress: order.deliveryAddress,
        deliveryTime: order.deliveryTime,
        paymentMethod: order.paymentMethod,
        deliveryMethod: order.deliveryMethod,
    }
}

File: src/ELibrary.Backend/ShopApi/Features/OrderFeature/Dtos/ClientUpdateOrderRequest.cs
using LibraryShopEntities.Domain.Entities.Shop;

namespace ShopApi.Features.OrderFeature.Dtos
{
    public class ClientUpdateOrderRequest
    {
        public int Id { get; set; }
        public string ContactClientName { get; set; }
        public string ContactPhone { get; set; }
        public string DeliveryAddress { get; set; }
        public DateTime DeliveryTime { get; set; }
        public PaymentMethod PaymentMethod { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
    }
}